search/includes/classes/class-health.php

```diff
@@ -176,44 +193,42 @@ public static function validate_index_posts_content( $start_post_id = 1, $last_p
 		}
 
 		do {
-			$next_batch_post_id = $start_post_id + self::CONTENT_VALIDATION_BATCH_SIZE;
+			$next_batch_post_id = $start_post_id + $batch_size;
 
 			if ( $last_post_id < $next_batch_post_id ) {
 				$next_batch_post_id = $last_post_id + 1;
 			}
 
-			if ( $is_cli ) {
-				\WP_CLI::line( sprintf( 'Validating posts %d - %d', $start_post_id, $next_batch_post_id - 1 ) );
+			if ( $is_cli && ! $silent ) {
+				echo sprintf( 'Validating posts %d - %d', $start_post_id, $next_batch_post_id - 1 ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
 			}
 			
 			$result = self::validate_index_posts_content_batch( $indexable, $start_post_id, $next_batch_post_id );
 
 			if ( is_wp_error( $result ) ) {
-				$result = [
-					'entity' => $indexable->slug,
-					'start_post_id' => $start_post_id,
-					'error' => $result->get_error_message(),
-				];
+				$result['errors'] = array( sprintf( 'batch %d - %d (entitiy: %s) error: %s', $start_post_id, $next_batch_post_id - 1, $indexable->slug, $result->get_error_message() ) );
 			}
 
 			$results = array_merge( $results, $result );
 
 			// Limit $results size
-			if ( strlen( serialize( $results ) ) > self::CONTENT_VALIDATION_MAX_DIFF_SIZE ) {
-				$error = new WP_Error( 'diff-size-limit-reached', sprintf( 'Reached diff size limit of %d bytes, aborting', self::CONTENT_VALIDATION_MAX_DIFF_SIZE ) );
+			if ( strlen( serialize( $results ) ) > $max_diff_size ) {
+				$error = new WP_Error( 'diff-size-limit-reached', sprintf( 'Reached diff size limit of %d bytes, aborting', $max_diff_size ) );
 
 				$error->add_data( $results, 'diff' );
 
 				return $error;
 			}
 
-			$start_post_id += self::CONTENT_VALIDATION_BATCH_SIZE;
+			$start_post_id += $batch_size; 
 
 			if ( $dynamic_last_post_id ) {
 				// Requery for the last post id after each batch b/c the site is probably growing
 				// while this runs
 				$last_post_id = self::get_last_post_id();
 			}
+
+			echo sprintf( "...%s\n", empty( $result ) ? '✅' : '❌' ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```

Comment:
Can we use `WP_CLI::line()` instead of echoing directly?

Reply:
If we're doing all the printing at the end rather than Validating posts %d - %d at the start and the status at the end.

Reply:
Oh right, WP_CLI::line() would send a full line, we just need to tack on extra info - should be fine.